R/S3.R

```diff
@@ -1077,151 +1085,75 @@ pander.zoo <- function(x, caption = attr(x, 'caption'), ...) {
 
 }
 
-
-#' Pander method for lme class
+#' Pander method for summary.lme class
 #'
 #' Prints a lme object in Pandoc's markdown.
-#' @param x a lme object
+#' @param xs a lme object
 #' @param caption caption (string) to be shown under the table
-#' @param summary (default:\code{FALSE}) if to print expender summary
+#' @param summary (default:\code{TRUE}) if to print expender summary
 #' @param ... optional parameters passed to raw \code{pandoc.table} function
 #' @export
-pander.lme <- function(x, caption = attr(x, 'caption'), summary = FALSE, ...) {
-
+pander.summary.lme <- function(xs, caption = attr(xs, 'caption'), summary = TRUE, ...) {
+    
     if (is.null(caption)) {
         if (is.null(storage$caption))
             caption <- sprintf('Linear mixed-effects model fit by %s : %s',
-                               paste(sub('^[ ]*', '', ifelse(x$method == 'REML', 'REML', 'maximum likelihood'))),
-                               pandoc.formula.return(x$call$fixed), collapse = '')
+                               paste(sub('^[ ]*', '', ifelse(xs$method == 'REML', 'REML', 'maximum likelihood'))),
+                               pandoc.formula.return(xs$call$fixed), collapse = '')
         else
             caption <- get.caption()
     }
-
-    xs  <- summary(x)
+    
     res <- as.data.frame(xs$tTable)
-
+    
     if (summary) {
-        pandoc.table(res, caption = pandoc.formula.return(x$call$fixed, text = 'Fixed effects: '),split.tables = Inf, ...)
-        pandoc.table(xs$residuals, caption = 'Standardized Within-Group Residuals')
+        pandoc.table(res, caption = pandoc.formula.return(xs$call$fixed, text = 'Fixed effects: '), split.tables = Inf, ...)
+        pandoc.table(xs$residuals, caption = 'Standardized Within-Group Residuals') 
         pandoc.table(data.frame(
-            'Observations'        = x$dims[["N"]],
-            'Groups'              = x$dims$ngrps[1:x$dims$Q],
-            'Log-restricted-likelihood' = x$logLik,
+            'Observations'        = xs$dims[["N"]],
+            'Groups'              = xs$dims$ngrps[1:xs$dims$Q],
+            'Log-restricted-likelihood' = xs$logLik,
             check.names = FALSE), keep.trailing.zeros = TRUE, caption = caption, digits = 4)
     } else {
-
+        
         pandoc.table(res, caption = caption, ...)
-
+        
```

Comment:
Can you please also remove the white space from the empty lines? My future commits would automatically revert this, so we might end up adding and removing white space all the time.